app/models/offer.rb

```diff
@@ -119,6 +119,15 @@ def scheduled?
     end
   end
 
+  def self.find_by_ggv_uuid(ggv_uuid)
+    begin
+      offer = GogovanOrder.find_by(ggv_uuid: ggv_uuid).delivery.offer
+    rescue Exception => e
+      raise ActiveRecord::RecordNotFound
+    end
+    with_eager_load.find(offer.id)
+  end
```

Comment:
@sunil-sharma  I think this method belongs on the GogovanOrder class as that is the primary class involved here. Also , as houndci specifies, it's not great to catch generic exceptions like this as we can't see what's really going wrong. It's fine for the API to return a 500 or a 404 - you can let it decide. So I would just remove the rescue entirely. If the lookup goes wrong, errbit will tell us and the user will get a 500 error - that's what we want.